### DISCUSSION TITLE
recompile Webpack

### DISCUSSION THREAD

**Discussion by bmilioli** [chatwoot/chatwoot#8037] [Q&A]
Hello everyone! I'm managing a self-hosted instance of Chatwoot and I'm in need of the exact commands to recompile Webpack. Could someone please provide the step-by-step commands or direct me to documentation that outlines the process within a self-hosted environment? Your assistance in sharing the specific Webpack recompilation commands would be greatly appreciated!

**Comment by jtayrone** [comment]
To recompile your instance, use the following commands:

```bash
# Login as Chatwoot user
sudo -i -u chatwoot

# Navigate to the Chatwoot directory
cd chatwoot

# Recompile the assets
rake assets:precompile RAILS_ENV=production

# Switch back to root user
exit

# Reload systemd files
systemctl daemon-reload

# Restart the chatwoot server
systemctl restart chatwoot.target
```

More details at [Upgrading to a newer version of Chatwoot ](https://www.chatwoot.com/docs/self-hosted/deployment/linux-vm#upgrading-to-a-newer-version-of-chatwoot)

**Reply by bmilioli** [reply]
Works! Thanks!
